app/src/main/java/com/futurumgame/base/factories/ores/CoalMine.java

```diff
@@ -13,7 +13,7 @@
 import java.util.LinkedList;
 import java.util.Queue;
 
-public class CoalMine extends Mine<Coal> {
+public final class CoalMine extends Mine<Coal> {
 
     private CoalMine() {
         super(CoalMine.class.getSimpleName(), Coal.factory(), new Units(1, 2));
```

Comment:
Why do you here use the CoalMine.class.getSimpleName() when you use a constant with the name everywhere else?